src/editor/mod.rs

```diff
@@ -100,13 +105,18 @@ impl<'arena, Node: Ast<'arena> + 'arena> Editor<'arena, Node> {
 
         for (node, tok) in self.tree.root().display_tokens(&self.format_style) {
             match tok {
-                DisplayToken::Text(s) => {
-                    // Hash the ref to decide on the colour
-                    let col = {
+                DisplayToken::Text(s, category) => {
+                    let col = if DEBUG_HIGHLIGHTING {
+                        // Hash the ref to decide on the colour
                         let mut hasher = DefaultHasher::new();
                         node.hash(&mut hasher);
                         let hash = hasher.finish();
                         cols[hash as usize % cols.len()]
+                    } else {
+                        *self
+                            .color_scheme
+                            .get(category)
+                            .unwrap_or(&Color::LIGHT_MAGENTA)
```

Comment:
In the future we may want to log unknown categories and ideally make sure each is logged only once. It'll make it easy to discover missing configuration or typos.

Reply:
Oh good point - I'll fix that now.